### ISSUE TITLE
Workbench crashes on newer Gnome distributions [CARROT-1058]

### ISSUE THREAD

**Issue by dweiss** [carrot2/carrot2#1254]
Seems to be related to this.

https://bugs.eclipse.org/bugs/show_bug.cgi?id=404776

---
Issue: CARROT-1058 (migrated from JIRA), created by Dawid Weiss (@dweiss), resolved Feb 20 2015
Attachments: [hs_err_pid18890.log](https://carrotsearch.github.io/jira-archive-public/attachments/carrot2/CARROT-1058/hs_err_pid18890.log), [hs_err_pid2364.log](https://carrotsearch.github.io/jira-archive-public/attachments/carrot2/CARROT-1058/hs_err_pid2364.log)
Sub-tasks:
 - CARROT-1084
- CARROT-1085
- CARROT-1086
- CARROT-1087
- CARROT-1088
- CARROT-1089

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Having spent three hours trying to upgrade to the newest RCP release, I have to postpone this issue for now. Please use a Windows system or MacOSX (or a virtual machine with an older Ubuntu) for now.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

FYI. An update of RCP/ Eclipse to 3.8.2 (latest stable 3.8.x) doesn't solve the problem.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

I've updated the code to use Eclipse 4.3.2 only to find out that it opens on Ubuntu 14.04... but crashes a bit later with:

```Java
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f23061c1ca9, pid=14946, tid=139789396182784
#
# JRE version: OpenJDK Runtime Environment (7.0_55-b14) (build 1.7.0_55-b14)
# Java VM: OpenJDK 64-Bit Server VM (24.51-b03 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libwebkitgtk-1.0.so.0+0x461ca9]  webkit_web_view_get_type+0x2489
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/dweiss/Downloads/carrot2-workbench-3.10.0-SNAPSHOT/hs_err_pid14946.log
#
# If you would like to submit a bug report, please include
# instructions on how to reproduce the bug and visit:
#   http://icedtea.classpath.org/bugzilla
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```


There are workarounds on the web that downgrade Webkit, but it all seems very clumsy.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

The upgrade to 4.x also screws up the app layout, similar to this guy's post.
http://www.eclipse.org/forums/index.php/t/366870/

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Looking into this again.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Wb doesn't crash with RCP 4.4.1, but it looks kind of ugly and I've had some odd behavior in visualization resizing.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

There are still some problems with Ubuntu Unity's menu but they seem to be common to all Eclipse/ RCP apps; a workaround: run C2 with:

```Java
env UBUNTU_MENUPROXY=0 ./carrot2-workbench
```

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Seems to be working (more or less stable).

**Comment by dweiss** [comment]
Comment by Donal  Simmie (migrated from JIRA)

Hi Dawid,

I have tried the workaround you suggest (env UBUNTU_MENUPROXY=0 ./carrot2-workbench) but I still get the same error:

1. C  [libsoup-2.4.so.1+0x6c2a1]  soup_session_feature_detach+0x11

I have attached my log file. I am on Ubuntu 14.04 using workbench 3.9.4 and have installed libwebkitgtk as per the carrot Web page suggestions (libwebkitgtk-1.0-0)

Is there something else I need to do?

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Hi Donal,

Please try the snapshot build – 3.9.4 doesn't contain the bugfix.

http://download.carrot2.org/head/3.10.0-SNAPSHOT/

There is a launcher shell script for ubuntu so you can do:

```Java
./carrot2-workbench-ubuntu
```

and it should work fine (I've tried on a vm).

**Comment by dweiss** [comment]
Comment by Donal  Simmie (migrated from JIRA)

Looks good Dawid. Thank you.
